### PR TITLE
feat: allow service type NodePort

### PR DESCRIPTION
**Pull request by acouette**
Hello,

This PR allows to provision a service of type NodePort. Along with the associated nodePorts values for web and websecure ports.

### CODE REVIEW

File: templates/service.yaml [hunk 1]
```diff
@@ -8,14 +8,20 @@ metadata:
...
     targetPort: web
+    {{- if (and (eq .Values.service.type "NodePort") (not (empty .Values.ports.web.nodePort)))}}
+    nodePort: {{ .Values.ports.web.nodePort }}
+    {{- end }}
   - port: 443
     name: websecure
     targetPort: websecure
+    {{- if (and (eq .Values.service.type "NodePort") (not (empty .Values.ports.websecure.nodePort)))}}
+    nodePort: {{ .Values.ports.websecure.nodePort }}
+    {{- end }}
```
Comment (dtomcej) [inline templates/service.yaml]: needs newline

File: values.yaml [hunk 2]
```diff
@@ -6,11 +6,16 @@ image:
 deployment:
   replicas: 1
 #
+service:
```
Comment (dtomcej) [inline values.yaml]: Should be commented out by default

File: values.yaml [hunk 3]
```diff
@@ -6,11 +6,16 @@ image:
...
+  type: LoadBalancer
+#
 ports:
   web:
     port: 8000
+    #nodePort: 30000
```
Comment (dtomcej) [inline values.yaml]:
```suggestion
    # nodePort: 30000
```

File: values.yaml [hunk 4]
```diff
@@ -6,11 +6,16 @@ image:
...
   websecure:
     port: 8443
+    #nodePort: 30001
```
Comment (dtomcej) [inline values.yaml]:
```suggestion
    # nodePort: 30001
```

File: values.yaml (hunk 2 above)
Comment (acouette) [reply values.yaml]: I am not sure to understand: if you comment the service section by default, the service won't have any type. Could you elaborate ?

Comment (dtomcej) [reply values.yaml]: The service should be commented out, as it should not be a requirement to set.

Please see https://github.com/containous/traefik-helm-chart/commit/08996411524c74d28895541a893116e5bc28eed0 as an example of what I mean about setting the default value if not set.

Comment (dtomcej) [reply values.yaml]: >Do you want me to add an if/else condition that is more resilient 

Yes please, that is preferable. Ideally we don't want to mandate more values in the values.yaml than is needed, as it ties our hands for backwards compatibility.

Comment (acouette) [reply values.yaml]: I tried to make it clean but the way helm implement the 'and' made me stack ifs.

**Pull request closed without merging**